### PR TITLE
Fix ob_end_flush() in JsSse init

### PR DESCRIPTION
**Pull request by Vladzimir**
https://gist.github.com/connerbw/a2fa7712efe135d5854f4d32d67ca09f#file-eventemitter-php-L114

### CODE REVIEW

File: src/JsSse.php [hunk 1]
```diff
@@ -190,8 +190,10 @@ protected function initSse(): void
         }
 
         // prevent buffering
-        if (ob_get_level()) {
+        $levels = ob_get_level();
+        for ($i = 0; $i < $levels; $i++) {
             ob_end_flush();
         }
+        flush();
```
Comment (mvorisek) [inline src/JsSse.php]: please explain why `flush()` is helpful

also note there is `self::flush()`

Comment (mvorisek) [inline src/JsSse.php:197]:
```suggestion
        while (ob_get_level() > 0) {
            ob_end_flush();
        }
        flush();
```

File: src/JsSse.php [hunk 2]
```diff
@@ -190,8 +190,9 @@ protected function initSse(): void
         }
 
         // prevent buffering
-        if (ob_get_level()) {
+        while (ob_get_level() > 0) {
             ob_end_flush();
```
Comment (mvorisek) [inline src/JsSse.php:194]: please make all tests passing

Comment (Vladzimir) [reply src/JsSse.php:194]: > please make all tests passing

Sorry, but I can't do this

File: src/JsSse.php (hunk 1 above)
Comment (Vladzimir) [reply src/JsSse.php]: > please explain why `flush()` is helpful

I don't know. The code is taken from the example I gave. If you think it is not needed, delete it.

File: src/JsSse.php (hunk 2 above)
Comment (mvorisek) [reply src/JsSse.php:194]: Obviously, we cannot merge anything that would break the whole CI :)

Might I know your usecase?

Comment (Vladzimir) [reply src/JsSse.php:194]: I use the framework https://github.com/f3-factory/fatfree-core And SSE stops working in it.
Or are you talking about something else?

Comment (mvorisek) [reply src/JsSse.php:194]: The fix seems legit. Upper layer/framework can open more than 1 output buffer. I even wonder why the atk4 tests passed when they rely on ob. In anycase, now the tests needs to be fixed to pass.

Comment (Vladzimir) [reply src/JsSse.php:194]: Unfortunately, I'm not good at testing.

**Comment by mkrecek234** [comment]
Hi there, I have the issue that output buffering is still active with any of the old or the new suggested code - I am running PHP-FPM on Apache, and it looks as if none of the code will stop the buffering. All scripts only transmit upon finish, and not in between. Any idea what to do in that situation with FPM?

**Comment by mvorisek** [comment]
@mkrecek234 open a separate issue for that and link https://github.com/php/php-src/pull/12785#discussion_r1412054712 there. This PR should be solely for the `ob_end_flush()` unroll as it is not easy to pass testing on itself.

**Comment by mvorisek** [comment]
we should fix it

File: src/JsSse.php (hunk 1 above)
Comment (mvorisek) [reply src/JsSse.php]: This was the cause of CI failure, as it prevented HTTP code to be set later.

Does this PR now fixes your issue with output buffering together with https://github.com/atk4/ui/pull/2175#discussion_r1501397309 framework you use?

File: src/JsSse.php (hunk 2 above)
Comment (mvorisek) [reply src/JsSse.php:194]: see https://github.com/atk4/ui/pull/2175#discussion_r1535566245 for the cause

**Pull request merged**